olp-cpp-sdk-dataservice-read/src/PrefetchPartitionsHelper.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <olp/core/client/CancellationContext.h>
+#include <olp/core/client/PendingRequests.h>
+#include <olp/core/logging/Log.h>
+#include <olp/dataservice/read/Types.h>
+
+#include "Common.h"
+#include "DownloadItemsJob.h"
+#include "ExtendedApiResponse.h"
+#include "ExtendedApiResponseHelpers.h"
+#include "QueryMetadataJob.h"
+#include "TaskSink.h"
+#include "repositories/PartitionsRepository.h"
+
+namespace olp {
+namespace dataservice {
+namespace read {
+
+using PartitionDataHandleResult =
+    std::vector<std::pair<std::string, std::string>>;
+using PartitionsDataHandleExtendedResponse =
+    ExtendedApiResponse<PartitionDataHandleResult, client::ApiError,
+                        client::NetworkStatistics>;
+
+class PrefetchPartitionsHelper {
+ public:
+  using DownloadJob = DownloadItemsJob<std::string, PrefetchPartitionsResult,
+                                       PrefetchPartitionsStatus>;
+  using QueryFunc = QueryItemsFunc<std::string, std::vector<std::string>,
+                                   PartitionsDataHandleExtendedResponse>;
+
+  static client::CancellationToken Prefetch(
+      std::shared_ptr<DownloadJob> download_job,
+      const std::vector<std::string>& roots, QueryFunc query,
+      TaskSink& task_sink, uint32_t query_max_size, uint32_t priority) {
```

Comment:
Sizes, i.e. query_max_size, are usually using type `size_t` and not `uint32_t`.
You even cast below and mostly use it together with other size_t vars.